### PR TITLE
Filter and DropMissing

### PR DESCRIPTION
**Pull request by eliascarv**
This PR add Filter and DropMissing transformations.

### CODE REVIEW

File: src/TableTransforms.jl [hunk 1]
```diff
@@ -30,6 +30,8 @@ export
   colapply, colrevert,
 
   # built-in
+  Filter,
+  DropMissing,
```
Comment (juliohm) [inline src/TableTransforms.jl:34]: Move to after Reject

File: src/transforms.jl [hunk 2]
```diff
@@ -211,6 +211,7 @@ end
 # IMPLEMENTATIONS
 # ----------------
 
+include("transforms/filter.jl")
```
Comment (juliohm) [inline src/transforms.jl:214]: Move to after select.jl

File: src/transforms/filter.jl [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+end
+
+isrevertible(::Type{<:Filter}) = true
+
+rowvalues(row, colnames) = [Tables.getcolumn(row, col) for col in colnames]
+
+function apply(transform::Filter, table)
+  tablerows = Tables.rows(table)
+  colnames = Tables.columnnames(table)
+  rows = [rowvalues(row, colnames) for row in tablerows]
```
Comment (juliohm) [inline src/transforms/filter.jl:21]: If we need to collect the rows anyways, then we can simply use the Tables.rowtable helper function.

File: src/transforms/filter.jl [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  rinds = Vector{Int}()
+  
+  for (i, row) in enumerate(rows)
+    if transform.func(row)
+      push!(srows, row)
+    else
+      push!(rrows, row)
+      push!(rinds, i)
+    end
+  end
```
Comment (juliohm) [inline src/transforms/filter.jl:35]: This could be replaced by a findall or filter call with arrays of tuples.

**Review by juliohm — CHANGES_REQUESTED** [review]

File: src/transforms.jl (hunk 2 above)
Comment (eliascarv) [reply src/transforms.jl:214]: OK

File: src/transforms/filter.jl (hunk 3 above)
Comment (eliascarv) [reply src/transforms/filter.jl:21]: OK, I'll do it.

File: src/transforms/filter.jl (hunk 4 above)
Comment (eliascarv) [reply src/transforms/filter.jl:35]: OK, I'll do it.

File: test/runtests.jl [hunk 5]
```diff
@@ -27,6 +27,12 @@ end
 Polynomial(args::T...) where {T<:Real} = Polynomial(collect(args))
 (p::Polynomial)(x) = sum(a * x^(i-1) for (i, a) in enumerate(p.coeffs))
 
+function testmissing(vec1, vec2)
+  r1 = map(!ismissing, vec1) == map(!ismissing, vec2)
+  r2 = findall(ismissing, vec1) == findall(ismissing, vec2)
+  r1 && r2
```
Comment (juliohm) [inline test/runtests.jl:33]: Can you elaborate on why we need these two checks instead of just the last with findall?

**Comment by juliohm** [comment]
As usual a great PR 👍🏽 we just need to double check the testmissing helper function, maybe we could avoid introducing a function if all we need is to check the positions of the missing values with findall.

File: test/runtests.jl (hunk 5 above)
Comment (eliascarv) [reply test/runtests.jl:33]: This test pass without first check:
```julia
[missing, 1, 2] == [missing, 2, 1]
```
And this test pass without second check:
```julia
[missing, 1, 2] == [1, 2, missing]
```

Comment (juliohm) [reply test/runtests.jl:33]: Thanks, I guess we could rewrite this helper function with a for loop like in the previous PR where we first check if the two entries are not missing and then compare the values?

**Comment by juliohm** [comment]
Also, we need to add these two new transforms to the README table 👍🏽

File: test/runtests.jl (hunk 5 above)
Comment (eliascarv) [reply test/runtests.jl:33]: Sorry, I messed up. Is no `map` is `filter`.

Comment (eliascarv) [reply test/runtests.jl]: Resolved with [0a3319d](https://github.com/JuliaML/TableTransforms.jl/pull/28/commits/0a3319d91a86cd4310f326976725eb82f7fadba2)

**Comment by eliascarv** [comment]
This PR is ready for a final review @juliohm!

File: README.md [hunk 6]
```diff
@@ -147,6 +147,8 @@ Please check the docstrings for additional information.
 |-----------|-------------|
 | `Select` | Column selection |
 | `Reject` | Column rejection |
+| `Filter` | Filter rows |
```
Comment (juliohm) [inline README.md:150]:
```suggestion
| `Filter` | Row filtering |
```

File: README.md [hunk 7]
```diff
@@ -147,6 +147,8 @@ Please check the docstrings for additional information.
...
+| `DropMissing` | Drop rows with missings |
```
Comment (juliohm) [inline README.md:151]:
```suggestion
| `DropMissing` | Drop missings |
```

**Review by juliohm — COMMENTED** [review]
I think we are ready for merging. I will adjust the descriptions in the README.

**Review by juliohm — APPROVED** [review]

**Pull request merged**